inginious/frontend/task_factory.py

```diff
@@ -241,6 +241,10 @@ def _get_last_updates(self, course, taskid, task_fs, need_content=False):
         if need_content:
             try:
                 task_content = descriptor_reader.load(task_fs.get(descriptor_name))
+                dispenser_data = course.get_descriptor()["dispenser_data"]
```

Comment:
The task should not be aware of the weight anymore. You can break any form of retrocompatibility here as ensuring it will most probably lead to a mess in the future. A migration script should be provided (in the future) to automatically move the task weights from task YAMLs to course YAMLs.